### PR TITLE
Convert remaining spring-rabbit tests to JUnit 5

### PR DESCRIPTION
**Pull request by garyrussell**

### CODE REVIEW

**Comment by artembilan** [comment]
Gary, any ideas about failure on Travis?
```
org.springframework.amqp.rabbit.listener.SimpleMessageListenerContainerIntegrationTests > testConsumerBatching(int, int, AcknowledgeMode, boolean, int, boolean)[11] FAILED

    java.lang.AssertionError at SimpleMessageListenerContainerIntegrationTests.java:280
```

File: build.gradle [hunk 1]
```diff
@@ -124,19 +124,18 @@ subprojects { subproject ->
 	// dependencies that are common across all java projects
 	dependencies {
 		compileOnly "com.google.code.findbugs:jsr305:$googleJsr305Version"
-		testCompile ("junit:junit:$junit4Version") {
```
Comment (artembilan) [inline build.gradle]: I guess we don't need `unit-vintage-engine` as well anymore...

File: build.gradle [hunk 2]
```diff
@@ -124,19 +124,18 @@ subprojects { subproject ->
...
-			exclude group: 'org.hamcrest', module: 'hamcrest-core'
-		}
 		testCompile "org.apache.logging.log4j:log4j-core:$log4jVersion"
 		testCompile "org.hamcrest:hamcrest-all:$hamcrestVersion"
 		testCompile ("org.mockito:mockito-core:$mockitoVersion") {
 			exclude group: 'org.hamcrest', module: 'hamcrest-core'
 		}
+		testCompile "org.mockito:mockito-junit-jupiter:$mockitoVersion"
```
Comment (artembilan) [inline build.gradle]: What is cool in this? Why do we need this extra dep?
Thanks

Comment (garyrussell) [reply build.gradle]: `@ExtendWith(MockitoExtension.class)` (for classes with `@Mock`, `@Spy`.

**Pull request merged**